src/main/kotlin/no/nav/henvendelsesarkiv/Database.kt

```diff
@@ -0,0 +1,93 @@
+package no.nav.henvendelsesarkiv
+
+import no.nav.henvendelsesarkiv.model.Arkivpost
+import no.nav.henvendelsesarkiv.model.ArkivpostMapper
+import no.nav.henvendelsesarkiv.model.Vedlegg
+import no.nav.henvendelsesarkiv.model.VedleggMapper
+import org.springframework.jdbc.core.JdbcTemplate
+
+private const val ARKIVPOST_SQL = """
+            INSERT INTO arkivpost(arkivpostId, arkivertDato, mottattDato, utgaarDato, temagruppe, arkivpostType, dokumentType,
+                                  kryssreferanseId, kanal, aktoerId, fodselsnummer, navIdent, innhold, journalfoerendeEnhet,
+                                  status, kategorikode, signert, erOrganInternt, begrensetPartInnsyn, sensitiv
+            ) VALUES (
+                ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?
+            )
+            """
+
+private const val VEDLEGG_SQL = """
+            INSERT INTO vedlegg(
+                arkivpostId, filnavn, filtype, variantformat, tittel, brevkode, strukturert, dokument
+            ) VALUES (
+                ?, ?, ?, ?, ?, ?, ?, ?
+            )
+            """
+
+class Database constructor(private val jt: JdbcTemplate, private val useHsql: Boolean = false) {
```

Comment:
Hva gjør de to parameterene?

Reply:
JdbcTemplate er SQL-hjelpe-klassen. Den andre sier om man bruker hsql (in-memory base til testing). Den siste parameteren må brukes i.f.m. tester fordi HSQL og Oracle (som er det som ligger i bunnen i prod) tolker uthenting av sekvens på ulike måter.